packages/merge-on-green/test/merge-on-green.ts

```diff
@@ -194,658 +122,110 @@ describe('merge-on-green', () => {
-          const pr = [
-            [
-              {
-                repo: 'testRepo',
-                number: 1,
-                owner: 'testOwner',
-                created: Date.now(),
-                branchProtection: ['Special Check'],
-                label: 'automerge: exact',
-                author: 'testOwner',
-                reactionId: 1,
-              },
-            ],
-          ];
-          return pr;
-        };
+      scopes.forEach(s => s.done());
+    });
 
-        const scopes = [
-          getRateLimit(5000),
-          getLatestCommit([{sha: '6dcb09b5b57875f334f61aebed695e2e4193db5e'}]),
-          getReviewsCompleted([
-            {
-              user: {login: 'octocat'},
-              state: 'APPROVED',
-              commit_id: '12345',
-              id: 12345,
-            },
-            {
-              user: {login: 'octokitten'},
-              state: 'APPROVED',
-              commit_id: '12346',
-              id: 12346,
-            },
-          ]),
-          dismissReview(12345),
-          dismissReview(12346),
-          getStatusi('6dcb09b5b57875f334f61aebed695e2e4193db5e', [
-            {state: 'success', context: 'Special Check'},
-          ]),
-          getCommentsOnPr([]),
-        ];
+    it('adds a PR if branch protection when PR labeled', async () => {
+      const addPRStub = sandbox.stub(handler, 'addPR').callsFake(async () => {
+        await handler.checkForBranchProtection(
+          'testOwner',
+          'testRepo',
+          1,
+          testingOctokitInstance
+        );
+      });
+      const scopes = [getBranchProtection(200, ['Special Check'])];
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const payload = require(resolve(
+        fixturesPath,
+        'events',
+        'pull_request_labeled'
+      ));
+      await probot.receive({
+        name: 'pull_request',
+        payload,
+        id: 'abc123',
+      });
 
-        await probot.receive({
-          name: 'schedule.repository' as '*',
-          payload: {org: 'testOwner'},
-          id: 'abc123',
-        });
+      scopes.forEach(s => s.done());
+      addPRStub.restore();
+    });
 
-        scopes.forEach(s => s.done());
+    //This function is supposed to respond with an error
+    it('does not add a PR if branch protection errors and comments on PR when PR labeled', async () => {
+      loggerStub.restore();
+
+      const scopes = [getBranchProtection(400, []), commentOnPR()];
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const payload = require(resolve(
+        fixturesPath,
+        'events',
+        'pull_request_labeled'
+      ));
+
+      await probot.receive({
+        name: 'pull_request',
+        payload,
+        id: 'abc123',
       });
+
+      scopes.forEach(s => s.done());
     });
   });
-
-  describe('merge-on-green wrapper logic', () => {
+  suite('stubbing all GCP functions', () => {
```

Comment:
This seems like a good middle-ground, compared to breaking out the tests into multiple files, or completely restructuring tests.